enterprise/backend/src/metabase_enterprise/database_routing/middleware.clj

```diff
@@ -33,14 +34,17 @@
 (defenterprise attach-destination-db-middleware
   "Pre-processing middleware. Calculates the destination database that should be used for this query, e.g. for caching
   purposes. Does not make any changes to the query besides (possibly) adding a `:destination-database/id` key."
-  :feature :database-routing
+  ;; run this even when the `:database-routing` feature is not enabled, so that we can assert that it *is* enabled if
+  ;; DB routing is configured. (Throwing here is better than silently ignoring the configured routing.)
```

Comment:
It's worth calling out a bit of a subtlety. We want to only throw in these features when the feature _is configured_. Lots of these things are in the style of "apply sandboxing if configured", and these get called on every query. It's only when we see something like "sandbox table 4 on field 17 = 'gadget'" that we want to assert that sandboxing is enabled, and _fail closed_ otherwise.